src/de.rs

```diff
@@ -70,15 +90,15 @@ impl<'a, 'de> de::Deserializer<'de> for &'a mut Depythonizer<'de> {
             self.deserialize_bool(visitor)
         } else if obj.is_instance::<PyByteArray>()? || obj.is_instance::<PyBytes>()? {
             self.deserialize_bytes(visitor)
-        } else if obj.is_instance::<PyDict>()? {
+        } else if P::Dict::is_instance(obj)? {
```

Comment:
I feel like this `is_instance`, and the check below for `P::List` shouldn't be necessary.

Especially for `Py::List`, I think we can instead do

`if let Ok(seq) = obj.downcast::<&PySeqence>()?`

...

and I think we should be using a `PyMapping` type, which doesn't yet exist in PyO3 but could easily be added.

Reply:
It is needed for making custom is_instance which incorporates both `Dict` and `MyDict`. I guess naming is a bit broken

Reply:
The point I'm trying to make is that if we use `PySequence` and the proposed `PyMapping` we can support both `dict` and user-provided mappings without having to ever use the `PythonizeTypes`.

Reply:
That seems like a nice fit, that would also remove lots of methods, but there is a problem:
https://docs.rs/pyo3/0.14.4/pyo3/class/sequence/trait.PySequenceProtocol.html

PyList doesn't implement sequence protocol right now, as well as PyDict doesnt implement mapping protocol.

Reply:
`PySequenceProtocol` is a different trait and not relevant here. You should use https://docs.rs/pyo3/0.14.4/pyo3/types/struct.PySequence.html, which is a dynamic type which can interact with all Python types with sequence-based indexing.

Reply:
Thanks for update. May be it would be better to wait for this pr to be finalized?
https://github.com/PyO3/pyo3/pull/1844

After its finalization, i can finalize it afterwards. Is it ok for you?

Reply:
Sure thing. If you like we can temporarily make `pythonize` depend on PyO3's `main` branch in git, because it may be a while until pyo3 0.15 is released.

Reply:
Yes, that would be helpful!